tests/unit/python/foglamp/services/core/api/test_auth_optional.py

```diff
@@ -0,0 +1,562 @@
+        request_data = {"username": "ajtest", "password": "F0gl@mp"}
+        with patch.object(middleware._logger, 'info') as patch_logger_info:
+            with patch.object(auth, 'is_valid_role', return_value=True) as patch_role:
+                with patch.object(User.Objects, 'get', return_value={'role_id': '2', 'uname': 'ajtest', 'id': '2'}) as patch_get_user:
+                    with patch.object(auth._logger, 'warning') as patch_logger_warning:
+                        resp = await client.post('/foglamp/user', data=json.dumps(request_data))
+                        assert 409 == resp.status
+                        assert 'User with the requested username already exists' == resp.reason
+                    patch_logger_warning.assert_called_once_with('Can not create a user, username already exists')
+                args, kwargs = patch_get_user.call_args
+                assert {'username': 'ajtest'} == kwargs
+            patch_role.assert_called_once_with(2)
+        patch_logger_info.assert_called_once_with('Received %s request for %s', 'POST', '/foglamp/user')
+
+    async def test_create_user(self, client):
+        data = {'id': '3', 'uname': 'ajtest', 'role_id': '2'}
+        expected = {}
+        expected.update(data)
+        request_data = {"username": "ajtest", "password": "F0gl@mp"}
+        ret_val = {"response": "inserted", "rows_affected": 1}
+        msg = 'User has been created successfully'
+        with patch.object(middleware._logger, 'info') as patch_logger_info:
+            with patch.object(auth, 'is_valid_role', return_value=True) as patch_role:
+                with patch.object(User.Objects, 'get', side_effect=[User.DoesNotExist, data]):
+                    with patch.object(User.Objects, 'create', return_value=ret_val) as patch_create_user:
+                            with patch.object(auth._logger, 'info') as patch_audit_logger_info:
+                                resp = await client.post('/foglamp/user', data=json.dumps(request_data))
+                                assert 200 == resp.status
+                                r = await resp.text()
+                                actual = json.loads(r)
+                                assert msg == actual['message']
+                                assert expected['id'] == actual['user']['userId']
+                                assert expected['uname'] == actual['user']['userName']
+                                assert expected['role_id'] == actual['user']['roleId']
+                            patch_audit_logger_info.assert_called_once_with(msg)
+                    patch_create_user.assert_called_once_with(request_data['username'], request_data['password'], int(expected['role_id']))
+            patch_role.assert_called_once_with(int(expected['role_id']))
+        patch_logger_info.assert_called_once_with('Received %s request for %s', 'POST', '/foglamp/user')
+
+    async def test_create_user_exception(self, client):
```

Comment:
What does this test cover? 500 exception - "'type' object is not subscriptable".
API is not handling and generic exception.

Reply:
`with patch.object(User.Objects, 'create', return_value=Exception) as patch_create_user:`